libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ProbabilityDistributionBoxChart.tsx

```diff
@@ -7,49 +7,65 @@ import {
   calculateBoxPlotDataFromErrorCohort,
   defaultModelAssessmentContext,
   ErrorCohort,
+  IHighchartBoxData,
   ModelAssessmentContext
 } from "@responsible-ai/core-ui";
 import { localization } from "@responsible-ai/localization";
 import { PointOptionsObject } from "highcharts";
+import _ from "lodash";
 import React from "react";
 
 interface IProbabilityDistributionBoxChartProps {
+  boxPlotState: IProbabilityDistributionBoxChartState;
   selectedCohorts: ErrorCohort[];
   probabilityOption?: IChoiceGroupOption;
+  onBoxPlotStateUpdate: (
+    boxPlotState: IProbabilityDistributionBoxChartState
+  ) => void;
 }
 
-class IProbabilityDistributionBoxChartState {}
+export interface IProbabilityDistributionBoxChartState {
+  boxPlotData: Array<IHighchartBoxData | undefined>;
+  outlierData: number[][] | undefined;
+}
 
-export class ProbabilityDistributionBoxChart extends React.Component<
-  IProbabilityDistributionBoxChartProps,
-  IProbabilityDistributionBoxChartState
-> {
+export class ProbabilityDistributionBoxChart extends React.Component<IProbabilityDistributionBoxChartProps> {
   public static contextType = ModelAssessmentContext;
   public context: React.ContextType<typeof ModelAssessmentContext> =
     defaultModelAssessmentContext;
 
+  public componentDidUpdate(
+    prevProps: IProbabilityDistributionBoxChartProps
+  ): void {
+    if (
+      this.props.boxPlotState.boxPlotData.length === 0 ||
+      !_.isEqual(prevProps.selectedCohorts, this.props.selectedCohorts) ||
+      !_.isEqual(
+        prevProps.probabilityOption!.id,
```

Comment:
In which case is probabilityOption undefined? do we have values in this always? If yes, can we make it mandatory param?

Reply:
Good point. From what I know, this is always defined for classification scenarios, and ProbabilityDistributionBoxChart is only displayed for classification scenarios. But this needs to double check with Roman, so I will just leave it as optional for now.